rasa/importers/autoconfig.py

```diff
@@ -0,0 +1,241 @@
+        f"# # No configuration for the NLU pipeline was provided. The following "
+        f"default pipeline was used to train your model.\n"
+        f"# # To customise it, uncomment and adjust the pipeline. See "
+        f"{DOCS_URL_PIPELINE} for more information.\n"
+    ),
+    "policies": (
+        f"# # No configuration for policies was provided. The following default "
+        f"policies were used to train your model.\n"
+        f"# # To customise them, uncomment and adjust the policies. See "
+        f"{DOCS_URL_POLICIES} for more information.\n"
+    ),
+}
+
+
+def get_configuration(config_file_path: Text) -> Dict[Text, Any]:
+    """Determine configuration from a configuration file.
+
+    Keys that are provided and have a value in the file are kept. Keys that are not
+    provided are configured automatically.
+
+    Args:
+        config_file_path: The path to the configuration file.
+    """
+    if config_file_path and os.path.exists(config_file_path):
+        config = io_utils.read_config_file(config_file_path)
+
+        missing_keys = _get_missing_config_keys(config)
+        keys_to_configure = _get_unspecified_autoconfigurable_keys(config)
+
+        if keys_to_configure:
+            config = _auto_configure(config, keys_to_configure)
+            _dump_config(config, config_file_path, missing_keys, keys_to_configure)
+
+    else:
+        logger.debug("No configuration file was provided to the TrainingDataImporter.")
+        config = {}
+
+    return config
+
+
+def _get_unspecified_autoconfigurable_keys(config: Dict[Text, Any]) -> List[Text]:
+    return [k for k in CONFIG_AUTOCONFIGURABLE_KEYS if not config.get(k)]
+
+
+def _get_missing_config_keys(config: Dict[Text, Any]) -> List[Text]:
+    return [k for k in CONFIG_KEYS if k not in config.keys()]
+
+
+def _auto_configure(
+    config: Dict[Text, Any], keys_to_configure: List[Text]
+) -> Dict[Text, Any]:
+    """Complete a config by adding automatic configuration for the specified keys.
+
+    Args:
+        config: The provided configuration.
+        keys_to_configure: Keys to be configured automatically (e.g. `policies`).
+
+    Returns:
+        The resulting configuration including both the provided and the automatically
+        configured keys.
+    """
+    import pkg_resources
+
+    if keys_to_configure:
+        logger.debug(
+            f"The provided configuration does not contain the key(s) "
+            f"{common_utils.transform_collection_to_sentence(keys_to_configure)}. "
+            f"Running automatic configuration for them now."
+        )
+
+    default_config_file = pkg_resources.resource_filename(
+        __name__, "default_config.yml"
+    )
+
+    default_config = io_utils.read_config_file(default_config_file)
+
+    config = copy.deepcopy(config)
+    for key in keys_to_configure:
+        config[key] = default_config[key]
+
+    return config
+
+
+def _dump_config(
+    config: Dict[Text, Any],
+    config_file_path: Text,
```

Comment:
optional: wouldn't it be easier to use `Path` here instead of working with strings?

Reply:
doesn't make much of a difference